app/jobs/health/process_enrollment_changes_job.rb

```diff
@@ -88,51 +88,58 @@ def perform(enrollment_id)
           disenrollment_date = Health::Enrollment.disenrollment_date(transaction)
           subscriber_id = Health::Enrollment.subscriber_id(transaction)
 
-          if disenrollment_date.present?
-            next if referral.nil? # This is a disenrollment, but we never enrolled this patient
-            next if referral.disenrolled? # This is a disenrollment, and the patient is already disenrolled
+          if disenrollment_date.present? && referral.nil?
+            # This is a disenrollment, but we don't have an enrollment
+            audit_actions[subscriber_id] = Health::Enrollment::DISENROLLMENT
+            referral = enroll_patient(transaction)
+            disenroll_patient(transaction, referral, file_date)
+            disenrolled_patients += 1
+
+          elsif disenrollment_date.present? && referral.disenrolled?
+            # This is a disenrollment, and the patient is already disenrolled
+            audit_actions[subscriber_id] = Health::Enrollment::DISENROLLMENT
+
+            if Health::Enrollment.enrollment_date(transaction) > referral.enrollment_start_date
+              # The audit is for an later enrollment than our current one, add it for consistency
+              referral = re_enroll_patient(referral, transaction)
+              disenroll_patient(transaction, referral, file_date)
+              disenrolled_patients += 1
+            end
 
+          elsif disenrollment_date.present?
             # This is a missed disenrollment
             audit_actions[subscriber_id] = Health::Enrollment::DISENROLLMENT
+
             disenroll_patient(transaction, referral, file_date)
             disenrolled_patients += 1
 
           elsif referral.nil?
             # This is a missed enrollment
             audit_actions[subscriber_id] = Health::Enrollment::ENROLLMENT
-            begin
-              enroll_patient(transaction)
-              new_patients += 1
-            rescue Health::MedicaidIdConflict
-              errors << conflict_message(transaction)
-            end
+
+            enroll_patient(transaction)
+            new_patients += 1
 
           elsif referral.disenrolled?
             # This is a missed re-enrollment
             audit_actions[subscriber_id] = Health::Enrollment::ENROLLMENT
+
             if referral.re_enrollment_blackout?(file_date)
               errors << blackout_message(transaction)
             else
-              begin
-                re_enroll_patient(referral, transaction)
-                returning_patients += 1
-              rescue Health::MedicaidIdConflict
-                errors << conflict_message(transaction)
-              end
```

Comment:
By moving this to the bottom, won't this stop if anyone in the batch has a conflict instead of catching each one?

Reply:
The rescue is in the block, so, execution will continue with the next audit record.